README.md

```diff
@@ -174,3 +175,108 @@ Explanation of the switches:
+
+
+# Adding new parsers
+
+## Introduction
+
+The extractor tool uses python classes as parser modules that can process specific formats as described above.
+
+In order to add a new custom parser, you have to write a parser class and add it to the directory in the following location: `o2r-meta\parsers`.
+
+Note that future versions of o2r-meta might be able to register all parsers based on the files in the `parsers` directory. Until then it is necessary to register a new parser in the function `register_parsers` of `o2r-meta\extract\metaextract.py`.
+
+The use of classes as parsers is currently still implemented in a rather naïve way. However, in order to be able to take advantage of non static methods, all parsers are designed as classes already. Also a governing class that provides inheritance features should be the goal of a better extractor.
+
+
+## Structure of your parser Python class
+
+_In this example we will call your new parser file `parse_abc.py` and the class in that file `ParseAbc`._
+
+Here is a extensively commented template for your parser file:
+
+```python
+# Class name keyword to signal availability to importing files
+__all__ = ['ParseAbc']
+
+# Optionally import the o2r-meta helping functions if want to have access to them
+from helpers.helpers import *
+
+# Import further external modules you would need and also add them to the requirements.txt file!
+import myABC
+
+# If the formats you want to parse depend on a successful import of external modules, use this
+try:
+    from myABC import *
+    FORMATS = ['.abc']
+except ImportError as iexc:
+    FORMATS = []
+    availability_issues = str(iexc)
+
+# The identifying name of your parser
+ID = 'o2r meta abc parser'
+
+
+# Declare your class
+class ParseAbc:
+	# Required method to return the ID
+    @staticmethod
+    def get_id():
+        return str(ID)
+
+	# Required method to return the available formats of this parser
+    @staticmethod
+    def get_formats():
+        if not FORMATS:
+        	# Status notes are available via the helpers module
+            status_note([__class__, ' unavailable (', str(availability_issues), ')'])
+        return FORMATS
+
+	# Required method. This is the place for your actual parsing code
+    @staticmethod
+    def parse(**kwargs):
+        is_debug = False
+        try:
+        	# The extractor provides the path of your target file
+            path_file = kwargs.get('p', None)
+            # And the metadata dictionary to which you want to write
+            MASTER_MD_DICT = kwargs.get('md', None)
+            is_debug = kwargs.get('is_debug', None)
+            # Fictional example to add parsed data
+            my_parsed_data = myABC.Dataset(path_file)
+            my_return = {path_file: {}}
+            my_return[path_file].update(my_parsed_data)
+            if 'ABC' in MASTER_MD_DICT:
+                if 'abc_files' in MASTER_MD_DICT['ABC']:
+                    MASTER_MD_DICT['ABC']['abc_files'].append(my_return)
```

Comment:
Where can I learn about the structure of `MASTER_MD_DICT` ?

Reply:
The question is: why would you need to know this? A new parser should not interfere with the existing keys in the MD dict, hence the new key `ABC` in this example.
(The structure of `MASTER_MD_DICT` is outlined in metaextract.py and the dummy file in the schema directory)

Reply:
If I should not interfere with existing keys, I must know the existing keys, don't I?

Your ()-sentence as a comment here would perfectly answer my question.

Reply:
> If I should not interfere with existing keys, I must know the existing keys, don't I?

I thought of this information as distracting but you're right in that the new parser must not use key names that are already in use.
I will add this